src/Gemstone.IO/Collections/FileBackedLookupTable.cs

```diff
@@ -1748,45 +1730,129 @@ static FileBackedLookupTable()
+                return null;
+
+            return (stream, obj) =>
+            {
+                using BinaryWriter writer = new BinaryWriter(stream, Encoding.UTF8, true);
+                binaryWriterAction(writer, obj);
+            };
         }
 
         private static Func<Stream, T>? GetReadMethod<T>()
         {
             Type type = typeof(T);
             ConstructorInfo? constructor = type.GetConstructor(Flags, null, s_types, null);
 
-            if (constructor is null)
+            if (constructor is not null)
             {
-                constructor = type.GetConstructor(Flags, null, Type.EmptyTypes, null);
+                List<ParameterExpression> parameterExpressions = s_types.Select(Expression.Parameter).ToList();
+                NewExpression newExpression = Expression.New(constructor, parameterExpressions);
+                LambdaExpression lambdaExpression = Expression.Lambda(typeof(Func<Stream, T>), newExpression, parameterExpressions);
+                return (Func<Stream, T>)lambdaExpression.Compile();
+            }
 
-                if (constructor is null)
-                    return null;
+            constructor = type.GetConstructor(Flags, null, Type.EmptyTypes, null);
 
+            if (constructor is not null)
+            {
                 MethodInfo? method = type.GetMethod("ReadFrom", Flags, null, s_types, null);
 
-                if (method is null)
-                    return null;
+                if (method is not null)
+                {
+                    return stream =>
+                    {
+                        T obj = Activator.CreateInstance<T>();
+                        method.Invoke(obj, new object[] { stream });
+                        return obj;
+                    };
+                }
+            }
+
+            Func<BinaryReader, T>? binaryReaderFunc = GetBinaryReaderMethod<T>();
 
+            if (binaryReaderFunc is not null)
+            {
                 return stream =>
                 {
-                    T obj = Activator.CreateInstance<T>();
-                    method.Invoke(obj, new object[] { stream });
-
-                    return obj;
+                    using BinaryReader reader = new BinaryReader(stream, Encoding.UTF8, true);
+                    return binaryReaderFunc(reader);
                 };
             }
 
-            List<ParameterExpression> parameterExpressions = s_types.Select(Expression.Parameter).ToList();
-            NewExpression newExpression = Expression.New(constructor, parameterExpressions);
-            LambdaExpression lambdaExpression = Expression.Lambda(typeof(Func<Stream, T>), newExpression, parameterExpressions);
+            return null;
+        }
+
+        private static Action<BinaryWriter, T>? GetBinaryWriterMethod<T>()
+        {
+            void WriteDateTime(BinaryWriter writer, DateTime dt)
+            {
+                writer.Write(dt.Ticks);
+                writer.Write((byte)dt.Kind);
```

Comment:
My nature here would be to encode the kind first, followed by the value such that when deserializing it would be possible to pass in deserialized value directly into `DateTime` constructor with known type. Generally speaking, type state parameters typically come first followed by the value, e.g., a length-prefixed string encoding.

Just a suggestion though...

Reply:
> ... such that when deserializing it would be possible to pass in deserialized value directly into `DateTime` constructor with known type.

I don't understand what you mean. You need both `ticks` and `kind` to invoke the appropriate `DateTime` constructor. If anything, the `kind` parameter would be omitted to invoke the constructor that implicitly assigns `DateTimeKind.Unspecified`. That suggests to me that `ticks` should come first since `kind` is optional. Although, it all seems moot because `ticks` and `kind` both need to be explicitly specified using this format because there's no bit reserved to specify whether `kind` is supplied.

I did them in this order based on the order of the constructor parameters. 🤷

Reply:
I meant order of serialization - not required parameters.

Reply:
Ahh, I see what you meant now.